include/integratorxx/quadratures/lebedev_laikov.hpp

```diff
@@ -5,241 +5,324 @@
-  using point_container  = typename base_type::point_container;
-  using weight_container = typename base_type::weight_container;
-  
-  LebedevLaikov(size_t npts): base_type( npts ) { }
+inline static int64_t algebraic_order_by_npts(int64_t npts) {
 
-  LebedevLaikov( const LebedevLaikov& ) = default;
-  LebedevLaikov( LebedevLaikov&& ) noexcept = default;
-};
+  switch (npts) {
+  case 6:
+    return 3;
+  case 14:
+    return 5;
+  case 26:
+    return 7;
+  case 38:
+    return 9;
+  case 50:
+    return 11;
+  case 74:
+    return 13;
+  case 86:
+    return 15;
+  case 110:
+    return 17;
+  case 146:
+    return 19;
+  case 170:
+    return 21;
+  case 194:
+    return 23;
+  case 230:
+    return 25;
+  case 266:
+    return 27;
+  case 302:
+    return 29;
+  case 350:
+    return 31;
+  case 434:
+    return 35;
+  case 590:
+    return 41;
+  case 770:
+    return 47;
+  case 974:
+    return 53;
+  case 1202:
+    return 59;
+  case 1454:
+    return 65;
+  case 1730:
+    return 71;
+  case 2030:
+    return 77;
+  case 2354:
+    return 83;
+  case 2702:
+    return 89;
+  case 3074:
+    return 95;
+  case 3470:
+    return 101;
+  case 3890:
+    return 107;
+  case 4334:
+    return 113;
+  case 4802:
+    return 119;
+  case 5810:
+    return 131;
+  default:
+    return -1;
+  }
+}
 
+inline static int64_t next_algebraic_order(int64_t order) {
 
-template <typename RealType>
-struct quadrature_traits< LebedevLaikov<RealType> > {
+  if (order <= 3)
+    return 3;
+  else if (order <= 5)
+    return 5;
+  else if (order <= 7)
+    return 7;
+  else if (order <= 9)
+    return 9;
+  else if (order <= 11)
+    return 11;
+  else if (order <= 13)
+    return 13;
+  else if (order <= 15)
+    return 15;
+  else if (order <= 17)
+    return 17;
+  else if (order <= 19)
+    return 19;
+  else if (order <= 21)
+    return 21;
+  else if (order <= 23)
+    return 23;
+  else if (order <= 25)
+    return 25;
+  else if (order <= 27)
+    return 27;
+  else if (order <= 29)
+    return 29;
+  else if (order <= 31)
+    return 31;
+  else if (order <= 35)
+    return 35;
+  else if (order <= 41)
+    return 41;
+  else if (order <= 47)
+    return 47;
+  else if (order <= 53)
+    return 53;
+  else if (order <= 59)
+    return 59;
+  else if (order <= 65)
+    return 65;
+  else if (order <= 71)
+    return 71;
+  else if (order <= 77)
+    return 77;
+  else if (order <= 83)
+    return 83;
+  else if (order <= 89)
+    return 89;
+  else if (order <= 95)
+    return 95;
+  else if (order <= 101)
+    return 101;
+  else if (order <= 107)
+    return 107;
+  else if (order <= 113)
+    return 113;
+  else if (order <= 119)
+    return 119;
+  else
+    return 131;
+}
 
-  using point_type  = cartesian_pt_t<RealType>;
-  using weight_type = RealType;
+} // namespace detail::lebedev
 
-  using point_container  = std::vector< point_type >;
-  using weight_container = std::vector< weight_type >;
+} // namespace IntegratorXX
```

Comment:
This is in the wrong location and is what is causing the current compile errors for the unit tests. It needs to be at the end of the file to properly scope `LebedevLaikov`

Reply:
^^ FWIW, I tested this locally and it works